lib/doorkeeper/oauth/pre_authorization.rb

```diff
@@ -5,15 +5,17 @@ module OAuth
     class PreAuthorization
       include Validations
 
-      validate :response_type, error: :unsupported_response_type
-      validate :client, error: :invalid_client
-      validate :scopes, error: :invalid_scope
-      validate :redirect_uri, error: :invalid_redirect_uri
+      validate :client,                error: :invalid_client
+      validate :redirect_uri,          error: :invalid_redirect_uri
```

Comment:
Because with missing or invalid `client` and `redirect_uri` param, `AuthorizationsController` must not redirect client to `redirect_uri`. So I change the order of `validate`:  **Firstly, checked `client` and `redirect_uri` param**. This order makes sure that ErrorResponse is redirectable or not.